app/src/main/java/org/projectbuendia/client/ui/chart/PatientChartController.java

```diff
@@ -269,21 +269,21 @@ public void suspend() {
         }
     }
 
-    public void onXFormResult(int requestCode, int resultCode, Intent data) {
-        FormRequest request = popFormRequest(requestCode);
-        if (request == null) {
+    public void onXFormResult(final int requestCode, final int resultCode, final Intent data) {
+        final FormRequest request = popFormRequest(requestCode);
+         if (request == null) {
             LOG.e("Unknown form request code: " + requestCode);
             return;
         }
 
-        boolean shouldShowSubmissionDialog = (resultCode != Activity.RESULT_CANCELED);
-        String action = (resultCode == Activity.RESULT_CANCELED)
-            ? "form_discard_pressed" : "form_save_pressed";
-        Utils.logUserAction(action,
-            "form", request.formUuid,
-            "patient_uuid", request.patientUuid);
-        mOdkResultSender.sendOdkResultToServer(request.patientUuid, resultCode, data);
-        mUi.showFormSubmissionDialog(shouldShowSubmissionDialog);
+        final boolean isSubmissionCanceled = (resultCode == Activity.RESULT_CANCELED);
+        Utils.logUserAction(isSubmissionCanceled ? "form_discard_pressed" : "form_save_pressed",
+            "form", request.formUuid, "patient_uuid", request.patientUuid);
+        if(isSubmissionCanceled) return;
```

Comment:
Yeah, we weren't ... lol .. sorry about that ...
But, if possible, could you check that comment too ? I would like to know your opinion about that.

Getting back to the point you've commented, i had left that check inside `OdkResultSender.sendOdkResultToServer` too because i thought that this method could be called by other controllers. But i agree with you. It is useless now. I'll delete it. Thx